packages/git-extractor/src/routes/github/index.ts

```diff
@@ -176,18 +177,27 @@ export const compare = async (ctx: Context) => {
   if (includeContents) {
     const files = await Promise.all(
       comparison.files.map(
-        ({ additions, changes, contents_url, deletions, filename, status }) => {
+        ({ additions, changes, contents_url, deletions, filename, status, patch, sha }) => {
           return api.getContent(contents_url, token).then((content) => {
             const data = content.content;
             const buffer = Buffer.from(data, content.encoding);
 
+            let stringContent: string
+
+            // If patch it is a text file, if not it is a binary
+            if (patch) {
+              stringContent = buffer.toString("utf-8")
+            } else {
+              stringContent = rawGitUrl({ username, repo, branch: "" }, filename, sha)
```

Comment:
Do we know what happens if the repo is private? In that case we don't use `rawGitUrl` for the URL (because the repo is private) and we upload the file to our own server and use that URL. I reckon it would be hard to make the comparison with that...

If this doesn't work for private, I might have a solution. We could generate a sha from the file that we already have (which we can download) and compare that with the sha given here.